x-pack/plugins/security_solution/public/common/components/event_details/summary_view.tsx

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import React, { useMemo } from 'react';
+
+import {
+  EuiTitle,
+  EuiDescriptionList,
+  EuiSpacer,
+  EuiDescriptionListTitle,
+  EuiDescriptionListDescription,
+  EuiInMemoryTable,
+  EuiBasicTableColumn,
+} from '@elastic/eui';
+
+import { get, getOr } from 'lodash/fp';
+import styled from 'styled-components';
+import { TimelineEventsDetailsItem } from '../../../../common/search_strategy';
+import { FormattedFieldValue } from '../../../timelines/components/timeline/body/renderers/formatted_field';
+import * as i18n from './translations';
+import { BrowserFields } from '../../../../common/search_strategy/index_fields';
+import {
+  ALERTS_HEADERS_RISK_SCORE,
+  ALERTS_HEADERS_RULE,
+  ALERTS_HEADERS_SEVERITY,
+} from '../../../detections/components/alerts_table/translations';
+import {
+  IP_FIELD_TYPE,
+  SIGNAL_RULE_NAME_FIELD_NAME,
+} from '../../../timelines/components/timeline/body/renderers/constants';
+import { DESTINATION_IP_FIELD_NAME, SOURCE_IP_FIELD_NAME } from '../../../network/components/ip';
+import { LineClamp } from '../line_clamp';
+import { useRuleAsync } from '../../../detections/containers/detection_engine/rules/use_rule_async';
+
+interface SummaryRow {
+  title: string;
+  description: {
+    contextId: string;
+    eventId: string;
+    fieldName: string;
+    value: string;
+    fieldType: string;
+    linkValue: string | undefined;
+  };
+}
+type Summary = SummaryRow[];
+
+const fields = [
+  { id: 'signal.status' },
+  { id: '@timestamp' },
+  {
+    id: SIGNAL_RULE_NAME_FIELD_NAME,
+    linkField: 'signal.rule.id',
+    label: ALERTS_HEADERS_RULE,
+  },
+  { id: 'signal.rule.severity', label: ALERTS_HEADERS_SEVERITY },
+  { id: 'signal.rule.risk_score', label: ALERTS_HEADERS_RISK_SCORE },
+  { id: 'host.name' },
+  { id: 'user.name' },
+  { id: SOURCE_IP_FIELD_NAME, fieldType: IP_FIELD_TYPE },
+  { id: DESTINATION_IP_FIELD_NAME, fieldType: IP_FIELD_TYPE },
+];
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const StyledEuiInMemoryTable = styled(EuiInMemoryTable as any)`
+  .euiTableHeaderCell {
+    border: none;
+  }
+  .euiTableRowCell {
+    border: none;
+  }
+`;
+
+const getTitle = (title: SummaryRow['title']) => (
```

Comment:
can we wrap it with `React.memo` ?

Reply:
this is used as the renderer for inMemoryTable, it fails after wrapping it with React.memo